src/buffer.jl

```diff
@@ -213,7 +215,9 @@ function insertdata!(buf::Buffer, data::Ptr{UInt8}, nbytes::Integer)
     end
     datasize = buf.marginpos - datapos
     copyto!(buf.data, datapos + nbytes, buf.data, datapos, datasize)
-    GC.@preserve buf unsafe_copyto!(bufferptr(buf), data, nbytes)
+    for i in 0:nbytes-1
```

Comment:
Could you use `copyto!`?

Reply:
`copyto!` isn't defined for `TranscodingStreams.Memory`, but it does have `getindex`

Reply:
Also `copyto!` seems like it has special cases for handling aliasing. https://github.com/JuliaLang/julia/issues/50900
This isn't needed here.